packages/core/src/amazonqGumby/chat/controller/controller.ts

```diff
@@ -0,0 +1,277 @@
+            } else {
+                this.messenger.sendStaticTextResponse('no-project-found', message.tabID)
+            }
+        }
+        return []
+    }
+
+    private async formActionClicked(message: any) {
+        const typedAction = MessengerUtils.stringToEnumValue(ButtonActions, message.action as any)
+        switch (typedAction) {
+            case ButtonActions.CONFIRM_TRANSFORMATION_FORM:
+                await this.initiateTransformationOnModule(message)
+                break
+            case ButtonActions.CANCEL_TRANSFORMATION_FORM:
+                this.messenger.sendJobFinishedMessage(message.tabId, true, undefined)
+                break
+            case ButtonActions.CONFIRM_JAVA_HOME_FORM:
+                await this.prepareProjectForSubmission(message)
+                break
+            case ButtonActions.CANCEL_JAVA_HOME_FORM:
+                this.messenger.sendJobFinishedMessage(message.tabId, true, undefined)
+                break
+            case ButtonActions.VIEW_TRANSFORMATION_HUB:
+                await vscode.commands.executeCommand(GumbyCommands.FOCUS_TRANSFORMATION_HUB)
+                this.messenger.sendJobSubmittedMessage(message.tabId)
+                break
+            case ButtonActions.STOP_TRANSFORMATION_JOB:
+                await stopTransformByQ(transformByQState.getJobId(), CancelActionPositions.Chat)
+                this.messenger.sendJobFinishedMessage(message.tabId, true)
+                break
+            case ButtonActions.CONFIRM_START_TRANSFORMATION_FLOW:
+                this.messenger.sendCommandMessage({ ...message, command: GumbyCommands.CLEAR_CHAT })
+                await this.transformInitiated({ ...message, tabID: message.tabId })
+                break
+        }
+    }
+
+    // Any given project could have multiple candidate modules to transform --
+    // The user gets prompted to pick a specific one
+    private async initiateTransformationOnModule(message: any) {
+        const pathToModule: string = message.formSelectedValues['GumbyTransformModuleForm']
+        const fromJDKVersion: string = message.formSelectedValues['GumbyTransformJdkFromForm']
+        const toJDKVersion: string = message.formSelectedValues['GumbyTransformJdkToForm']
+
+        this.messenger.sendCompilationInProgress(message.tabId, true)
+
+        await processTransformFormInput(pathToModule, fromJDKVersion, toJDKVersion)
+        await this.validateBuildWithPromptOnError(message)
+    }
+
+    private async prepareProjectForSubmission(message: any | undefined = undefined): Promise<void> {
+        if (message !== undefined && message.formSelectedValues !== undefined) {
+            const javaHome: string = message.formSelectedValues['JavaHomeFormInput'].trim()
+
+            if (!javaHome) {
+                const errorMessage = 'No JDK path provided'
+                this.messenger.sendErrorMessage(errorMessage, message.tabID)
+                throw new JavaHomeNotSetError()
```

Comment:
Do we need a custom error class?

Reply:
Question: Can we use `AWSToolkitError here`?

Reply:
Ignore: I see `JavaHomeNotSetError` extends `ToolkitError`

Reply:
Yeah but still wondering in general why not just use ToolkitError directly. Something like `new ToolkitError("message", { code: "some code"})` or something

Reply:
I check the type of errors in a `try catch` block so that I can show relevant error messages in the chat. This is more personal preference - I find custom errors easier to read / handle